### ISSUE TITLE
Calling "enable sw mfa" only should not reset mfa until is confirmed

### ISSUE THREAD

**Issue by ujovlado** [keboola/developer-portal#185]
Calling https://kebooladeveloperportal.docs.apiary.io/#reference/0/auth/enable-software-mfa will reset MFA and user can log in without MFA.

How to simulate:

1. enable sw MFA
2. confirm sw MFA
3. log in (MFA code is required)
4. call enable sw MFA again (e.g. for setting up new device)
5. logout
6. log in (no MFA is required)

I think that old MFA setup should not be disabled until confirmation is done.

**Comment by JakubMatejka** [comment]
Tvl to asi dělá Cognito, já tam nic takovýho nevyvádím.

**Comment by ujovlado** [comment]
Vies mi pls potvrdit, ze sa ti to deje tiez? Pripadne, ak je to chyba Cognito, nereportneme im to?

**Comment by JakubMatejka** [comment]
Jo, děje se to, i když to celý nastavím přes Cognito cli. Jdu jim to sepsat na support

**Comment by JakubMatejka** [comment]
Nějak mi to vypadlo a nevšim jsem si, že odpověděli. Akorát tomu moc nerozumím. ;o) Když nemáme MFA povinný, tak se to děje proto, že tam máme to sms?

> Our service team looked into the issue.  This behavior is because MFA is Optional on your userpool.
> But while I did some testing to provide you with the best solution, I found some bugs with the Software MFA.  Currently the only scenario where Software MFA works is if you have both SMS and Software enabled on the pool.  And the only scenario you can have MFA always enabled is for it to be required on the pool and have both SMS and Software MFA enabled, and when you associate but not verify a software token it will fall back to SMS MFA until the Software token is verified.

**Comment by ujovlado** [comment]
Ja som vtedy tu ich odpoved cital, ak si dobre spominam. A asi sme sa o tom aj bavili. Vlastne nam napisali, ze mozeme mat fallback na SMS - to ale nechceme.

Imho tam maju bug, pretoze pokial som nove zariadanie nepotvrdil, nemoze mi predsa zrusit stare :)

**Comment by odinuv** [comment]
jestli to nefixnou, nemuzem to aspon nejak porest v UI? protoze ted jsem se fest vypekl, ze jsem kliknul na "Change Device" a krizek a zadavam kody a heleho oni najednou chodi SMSkou
